### PR TITLE
Put metric before label value in the "label value too long" error mes…

### LINKED ISSUE #1582: Error message for 'label value too long' misses important information
The error seen by sender in Prometheus logs is like this:

```
level=warn ts=2019-07-22T09:25:16.899437956Z caller=queue_manager.go:530 component=remote queue=0:https://cloud.weave.works./api/prom/push msg="Error sending samples to remote storage" count=1000 err="server returned HTTP status 400 Bad Request: label value too long: \"{\\\"apiVersion\\\":\\\"extensions/v1beta1\\\",\\\"kind\\\":\\\"DaemonSet\\\",\\\"metadata\\\":{\\\"annotations\\\":{\\\"flux.weave.works/sync-checksum\\\":\\\"0b6f1382edb3f01d6a64cd775b46008b0f61003e\\\"},\\\"labels\\\":{\\\"flux.weave.works/sync-gc-mark\\\":\" metric \"kub"
level=warn ts=2019-07-22T09:25:32.00067315Z caller=queue_manager.go:530 component=remote queue=0:https://cloud.weave.works./api/prom/push msg="Error sending samples to remote storage" count=1000 err="server returned HTTP status 400 Bad Request: label value too long: \"{\\\"apiVersion\\\":\\\"extensions/v1beta1\\\",\\\"kind\\\":\\\"Deployment\\\",\\\"metadata\\\":{\\\"annotations\\\":{\\\"flux.weave.works/automated\\\":\\\"true\\\",\\\"flux.weave.works/automated_msg\\\":\\\"test\\\",\\\"flux.weave.works/automated_user\\\":\\\"Maarten\" metric \""
```

It looks like something is truncating the message just as it gets to the `metric=...` part.  Maybe put that before the value?
And it would also be useful to state the label name.

### PR DESCRIPTION
**Pull request by ChinYing-Li**
…sage (#1582)

Signed-off-by: ChinYing-Li <chinying.li@mail.utoronto.ca>

**What this PR does**:
This PR is #3982 applied to `d45e65ad1e90d22dbdbe224f6eb4036847b56c82`.
Swap the position of the metric string and the label value string in error message of `labelValueTooLong`.
Now the metric string is placed before the label value string.

**Which issue(s) this PR fixes**:
Fixes #1582

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/util/validation/errors.go
Comment (pstibrany) [inline pkg/util/validation/errors.go:31]: `labelValueTooLongError` doesn't need to wrap `genericValidationError`, and acccess its inner fields. It can just define its own fields directly.

File: pkg/util/validation/errors.go
Comment (pstibrany) [inline pkg/util/validation/errors.go:51]: I'd suggest to remove `message` field completely, and put this message into `(*labelValueTooLongError).Error()` method (`fmt.Sprintf` call), so that the message and correct order of parameters are together.

**Review by pstibrany — APPROVED** [review]
Thank you!

File: pkg/util/validation/errors.go
Comment (pracucci) [inline pkg/util/validation/errors.go:37]:
```suggestion
	return fmt.Sprintf("label value too long for metric: %.200q label value: %.200q", formatLabelSet(e.series), e.cause)
```

**Comment by pracucci** [comment]
@ChinYing-Li Could you fix the linter, please?

File: pkg/util/validation/errors.go
Comment (pstibrany) [inline pkg/util/validation/errors.go:37]:
```suggestion
	return fmt.Sprintf("label value too long for metric: %.200q label value: %.200q", formatLabelSet(e.series), e.cause)
```

File: pkg/util/validation/errors.go
Comment (pstibrany) [inline pkg/util/validation/errors.go:32]: nit: I'd suggest to rename `cause` to `labelValue`.

**Review by pstibrany — APPROVED** [review]

**Comment by ChinYing-Li** [comment]
Sorry about the problem with linting! Just fixed that.

File: pkg/util/validation/errors.go
Comment (pstibrany) [inline pkg/util/validation/errors.go:37]: Let's please fix the message before merging:

```suggestion
	return fmt.Sprintf("label value too long for metric: %.200q label value: %.200q", formatLabelSet(e.series), e.labelValue)
```

File: CHANGELOG.md
Comment (pracucci) [inline CHANGELOG.md:22]: We're used to link the PR number.

```suggestion
* [ENHANCEMENT] Put metric before label value in the "label value too long" error message. #4018
```

File: pkg/util/validation/errors.go
Comment (pracucci) [inline pkg/util/validation/errors.go:38]: [nit] Please move this below, to keep it just above its usage in `newLabelValueTooLongError()`

Comment (pracucci) [reply pkg/util/validation/errors.go:38]: Anyway, it's not a big deal. We can live with that. I will merge once CI passes.

**Review by pracucci — APPROVED** [review]

**Pull request merged**
